src/Stack/Config.hs

```diff
@@ -270,6 +271,24 @@ configFromConfigMonoid
      configLocalProgramsBase <- case getFirst configMonoidLocalProgramsBase of
        Nothing -> getDefaultLocalProgramsBase configStackRoot configPlatform origEnv
        Just path -> return path
+     let localProgramsFilePath = toFilePath configLocalProgramsBase
+     when (osIsWindows && ' ' `elem` localProgramsFilePath) $ do
+       logInfo $ "Stack's 'programs' path contains a space character. " <>
```

Comment:
It seems a bit excessive to always print this message, given that on many systems this setup will work just fine. The warning below seems like it will address the actual problem: warning the user when there will almost certainly be some problem.

Reply:
I have revised the pull request to only include the warning (and to fix the CI failures).